README.md

```diff
@@ -141,14 +141,76 @@ This repository contains code for both infrastructure and applications. Configur
+- **Purpose**: This generates a changelog and bumps the version number.
+- **Merge**: Merge the release pull request into the `main` branch.
+
+#### 4. Deploy to Staging (Bump Version and Create Tag)
+- **Trigger**: Merging the release pull request.
+- **Action**: 
+  - Bumps the version number.
+  - Generates the release and changelog.
+  - Deployment is tagged with the new `<version>` without `<git-sha>`
+  - The new version is built and deployed to the staging environment.
+
+#### 5. Prepare deployment to Production
+- **Action**: Perform a dry run towards the production environment to ensure the deployment can proceed without issues.
+
+#### 6. Deploy to Production
+- **Trigger**: Approval of the dry run.
+- **Action**: The new version is built and deployed to the production environment.
+
+#### Visual Workflow
 
 ![Deployment process](docs/deploy-process.png)
 
 [Release Please](https://github.com/google-github-actions/release-please-action) is used in order to create releases, generate changelog and bumping version numbers.
 
 `CHANGELOG.md` and `version.txt` are automatically updated and should not be changed manually.
 
+### Manual deployment (⚠️ handle with care)
+
+This project utilizes two GitHub dispatch workflows to manage manual deployments: `dispatch-apps.yml` and `dispatch-infrastructure.yml`. These workflows allow for manual triggers of deployments through GitHub Actions, providing flexibility for deploying specific versions to designated environments.
+
+#### Using `dispatch-apps.yml`
+
+The `dispatch-apps.yml` workflow is responsible for deploying applications. To trigger this workflow:
+
+1. Navigate to the Actions tab in the GitHub repository.
+2. Select the `Dispatch Apps` workflow.
+3. Click on "Run workflow".
+4. Fill in the required inputs:
+   - **environment**: Choose the target environment (`test`, `staging`, or `prod`).
+   - **version**: Specify the version to deploy. Could be git tag or a docker-tag published in packages.
+   - **runMigration** (optional): Indicate whether to run database migrations (`true` or `false`).
+
+This workflow will handle the deployment of applications based on the specified parameters, ensuring that the correct version is deployed to the chosen environment.
+
+#### Using `dispatch-infrastructure.yml`
+
+The `dispatch-infrastructure.yml` workflow is used for deploying infrastructure components. To use this workflow:
+
+1. Go to the Actions tab in the GitHub repository.
+2. Select the `Dispatch Infrastructure` workflow.
+3. Click on "Run workflow".
+4. Provide the necessary inputs:
+   - **environment**: Select the environment you wish to deploy to (`test`, `staging`, or `prod`).
+   - **version**: Enter the version to deploy, which should correspond to a git tag.
+
+This workflow facilitates the deployment of infrastructure to the specified environment, using the version details provided.
+
 ### Github actions
 
 Naming conventions for github actions:
```

Comment:
```suggestion
### GitHub Actions

Naming conventions for GitHub Actions:
```

Not really in your PR, but just noticed they don't match the casing in the rest of the document 🙈